tutorials/k8s-kapsule-multi-az/index.mdx

```diff
@@ -0,0 +1,625 @@
+    }
+
+    provider "helm" {
+    kubernetes {
+        host                   = scaleway_k8s_cluster.kapsule_multi_az.kubeconfig[0].host
+        token                  = scaleway_k8s_cluster.kapsule_multi_az.kubeconfig[0].token
+        cluster_ca_certificate = base64decode(scaleway_k8s_cluster.kapsule_multi_az.kubeconfig[0].cluster_ca_certificate)
+    }
+    }
+    ```
+
+2. To configure the distribution of the pods on the cluster we will use labels set on nodes. To get the label list of a node, you can use the `kubectl` command below. You can see that multiple labels, sets by the scaleway-cloud-controller-manager, contain information about the `region` and `zone` where the node is deployed. The label `topology.kubernetes.io/zone`, at the end of the list, is particularly interesting in your case as it will allow us to know on which availability zone a pod will be deployed.
+    ```
+    kubectl get node scw-kapsule-multi-az-pool-fr-par-1-51c432554c4 --template '{{range $k, $v := .metadata.labels}}{{printf "%s=%s\n" $k $v}}{{end}}'
+
+    beta.kubernetes.io/arch=amd64
+    beta.kubernetes.io/instance-type=PRO2-XXS
+    beta.kubernetes.io/os=linux
+    failure-domain.beta.kubernetes.io/region=fr-par
+    failure-domain.beta.kubernetes.io/zone=fr-par-1
+    k8s.scaleway.com/kapsule=6ee0a129-ee3b-4a89-ad4b-a74754441746
+    k8s.scaleway.com/managed=true
+    k8s.scaleway.com/multi-az=
+    k8s.scaleway.com/node=51c43255-4c42-449a-9e63-d734c357cd65
+    k8s.scaleway.com/pool=2d669c67-bb46-4e4f-93a0-a59834f6c6ba
+    k8s.scaleway.com/pool-name=pool-fr-par-1
+    k8s.scaleway.com/runtime=containerd
+    kubernetes.io/arch=amd64
+    kubernetes.io/hostname=scw-kapsule-multi-az-pool-fr-par-1-51c432554c4
+    kubernetes.io/os=linux
+    node.kubernetes.io/instance-type=PRO2-XXS
+    topology.csi.scaleway.com/zone=fr-par-1
+    topology.kubernetes.io/region=fr-par
+    topology.kubernetes.io/zone=fr-par-1
+    ```
+
+3. There are multiple very important configurations here. You want to have one `ingress-nginx` pod per Availability Zone, so the first thing is to have `replicaCount` to `3`. Doing only that you do not have any guarantee that Kubernetes will dispatch the replicas on different zones, so we define a `topologySpreadConstraints` to tell Kubernetes scheduler to spread the replicas based on the label `topology.kubernetes.io/zone`, which have one of the three `fr-par-1`, `fr-par-2` and `fr-par-3` value. You can find the full documentation for the [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) on Kubernetes documentation. On the next step, we will create through Terraform your different Kubernetes services to deploy multiple LoadBalancer, so we deactivate the creation of the default service by the Helm chart. Finally, we had a short snippet to the Nginx configuration of your ingress-nginx, which exposes a `/up` path that we will use further in your DNS configuration.
```

Comment:
Can we split this whole block into sub pieces? Because (i) it's hard to read as is (ii) will be better structure for memorization

Suggest as:
- You want to have
- Doing only that
- On the next step
- Finally, we...

Reply:
```suggestion
3. You want to have one `ingress-nginx` pod per Availability Zone. To do so, set `replicaCount` to `3`. However, this alone won't guarantee distribution across zones. Next, use `topologySpreadConstraints` to instruct the Kubernetes scheduler to distribute replicas based on the `topology.kubernetes.io/zone` label, which can have values like `fr-par-1`, `fr-par-2`, or `fr-par-3`. You can find detailed documentation for [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/) in the Kubernetes documentation.

Then, proceed to create various Kubernetes services through Terraform to deploy multiple `LoadBalancers`, ensuring that the default service creation by the Helm chart is disabled.

Finally, add a short snippet to the Nginx configuration of your `ingress-nginx` to expose a `/up` path, which will be used in your DNS configuration later.
```

cc @Nox-404 wdyt?